pkg/astnormalization/inject_input_default_values.go

```diff
@@ -0,0 +1,111 @@
+package astnormalization
+
+import (
+	"github.com/buger/jsonparser"
+	"github.com/jensneuse/graphql-go-tools/pkg/ast"
+	"github.com/jensneuse/graphql-go-tools/pkg/astvisitor"
+)
+
+func injectInputFieldDefaults(walker *astvisitor.Walker) *inputFieldDefaultInjectionVisitor {
+	visitor := &inputFieldDefaultInjectionVisitor{
+		Walker:       walker,
+		variableTree: make([]string, 0),
+	}
+	walker.RegisterEnterDocumentVisitor(visitor)
+	walker.RegisterVariableDefinitionVisitor(visitor)
+	return visitor
+}
+
+type inputFieldDefaultInjectionVisitor struct {
+	*astvisitor.Walker
+
+	operation  *ast.Document
+	definition *ast.Document
```

Comment:
Just making sure you need both operation definition?

Reply:
yeah need the definition to check the types

Reply:
Gotcha